code/modules/modular_computers/file_system/programs/ntdownloader.dm

```diff
@@ -155,44 +156,77 @@
 	if(prog.downloaderror) // Download errored. Wait until user resets the program.
 		data["error"] = prog.downloaderror
 	else if(prog.downloaded_file) // Download running. Wait please..
+		if (ui)
+			ui.set_auto_update(1)
 		data["downloadname"] = prog.downloaded_file.filename
 		data["downloaddesc"] = prog.downloaded_file.filedesc
 		data["downloadsize"] = prog.downloaded_file.size
 		data["downloadspeed"] = prog.download_netspeed //Even if it does update every 2 seconds, this is bad coding on everyone's count. :ree:
 		data["downloadcompletion"] = round(prog.download_completion, 0.01)
 	else // No download running, pick file.
+		if (ui)
+			ui.set_auto_update(0)//No need for auto updating on the software menu
 		data["disk_size"] = my_computer.hard_drive.max_capacity
 		data["disk_used"] = my_computer.hard_drive.used_capacity
-		var/list/all_entries[0]
 		if(!my_computer.software_locked) //To lock installation of software on work computers until the IT Department is properly implemented
```

Comment:
I'm not even sure why this wasnt working before, it just magically started once i began altering things. Probably an indentation issue or somesuch